src/poll.rs

```diff
@@ -257,19 +259,15 @@ impl Poll {
     /// # }
     /// ```
     pub fn new() -> io::Result<Poll> {
-        is_send::<Poll>();
-        is_sync::<Poll>();
-
-        let selector = Arc::new(sys::Selector::new()?);
-
-        let registry = Registry { selector };
-
-        Ok(Poll { registry })
+        sys::Selector::new().map(|selector| Poll { selector })
     }
 
-    /// Return a reference to the associated `Registry`.
-    pub fn registry(&self) -> &Registry {
-        &self.registry
+    /// Create a separate `Registry` which can be used to register
+    /// `event::Source`s.
+    pub fn registry(&self) -> io::Result<Registry> {
```

Comment:
It should be possible to keep this function unchanged. `Poll` is a wrapper around `Registry`. Then, we can add a `try_clone(&self) -> io::Result<Registry>` function on `Registry`.

That would allow us to easily switch back to an internal `Arc` if we need to w/o a breaking change (add `Clone` impl to `Registry` and deprecate `try_clone()`).